src/common/util/src/file_util.cpp

```diff
@@ -403,29 +325,10 @@ void ov::util::create_directory_recursive(const std::string& path) {
     }
 }
 
-bool ov::util::directory_exists(const std::string& path) {
-    struct stat sb;
-
-    if (stat(path.c_str(), &sb) == 0 && S_ISDIR(sb.st_mode)) {
-        return true;
-    }
-    return false;
-}
-
-#ifdef OPENVINO_ENABLE_UNICODE_PATH_SUPPORT
-bool ov::util::directory_exists(const std::wstring& path) {
-#    ifdef _WIN32
-    struct stat sb;
-
-    if (wstat(path.c_str(), &sb) == 0 && S_ISDIR(sb.st_mode)) {
-        return true;
-    }
-    return false;
-#    else
-    return directory_exists(wstring_to_string(path));
-#    endif
+bool ov::util::directory_exists(const ov::util::Path& path) {
+    std::error_code ec;
+    return std::filesystem::is_directory(path, ec);
```

Comment:
```suggestion
    return std::filesystem::is_directory(std::filesystem::status(path));
```
or
```suggestion
    return std::filesystem::status(path) == std::filesystem::file_type::directory;
```
Should do the same without error code.